### PR TITLE
added excludePages and graceful-fs

### PR DESCRIPTION
**Pull request by prceasar**

### CODE REVIEW

File: packages/lastrev-build-prefetch-content/src/helpers/excludeContent.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  defaultLocale?: string
+): { excludeIfParentIsExcluded: boolean; exclude: boolean } => {
+  const contentTypeId = get(content, 'sys.contentType.sys.id');
+  const config = get(buildConfig, `excludePages.${contentTypeId}`);
+
+  if (!contentTypeId || !config) return { excludeIfParentIsExcluded: false, exclude: false };
+
+  const { rules, fields, excludeIfParentIsExcluded } = config;
+
+  const data = chain(fields)
```
Comment (spindle79) [inline packages/lastrev-build-prefetch-content/src/helpers/excludeContent.ts:20]: If `fields` is undefined/null will this fail?

File: packages/lastrev-build-prefetch-content/src/helpers/excludeContent.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    .keyBy(identity)
+    .mapValues((fieldName) => getFieldValue(content, fieldName, locale, defaultLocale))
+    .value();
+
+  const exclude = !!jsonLogic.apply(rules, data);
```
Comment (spindle79) [inline packages/lastrev-build-prefetch-content/src/helpers/excludeContent.ts:25]: If `rules` or `data` is undefined/null will this fail?

File: packages/lastrev-build-prefetch-content/src/types.ts [hunk 3]
```diff
@@ -103,13 +104,22 @@ export type SettingsConfig = {
   include?: number;
 };
 
+export type ExlcudePagesConfig = {
```
Comment (spindle79) [inline packages/lastrev-build-prefetch-content/src/types.ts:107]: misspelling

File: packages/lastrev-build-prefetch-content/src/types.ts [hunk 4]
```diff
@@ -103,13 +104,22 @@ export type SettingsConfig = {
...
+};
+
 type OptionsBuildConfig = {
   excludeTypes: string[];
   mappings: MappingConfig;
   paths: PathsConfig;
   contentJson: ContentJsonConfig;
   settings: SettingsConfig;
   locales: LocalesConfig;
+  excludePages: ExlcudePagesConfig;
```
Comment (spindle79) [inline packages/lastrev-build-prefetch-content/src/types.ts:122]: misspelling

**Review by spindle79 — COMMENTED** [review]
Couple questions in the excludeContent.ts file

File: packages/lastrev-build-prefetch-content/src/helpers/excludeContent.ts (hunk 1 above)
Comment (prceasar) [reply packages/lastrev-build-prefetch-content/src/helpers/excludeContent.ts:20]: No. lodash will handle that and just returns `{}`

File: packages/lastrev-build-prefetch-content/src/helpers/excludeContent.ts (hunk 2 above)
Comment (prceasar) [reply packages/lastrev-build-prefetch-content/src/helpers/excludeContent.ts:25]: based on comment above, data will never be undefined. rules could be though, but json-logic also handles this without erroring. `!!` at the begining will ensure that that returns false.

**Pull request merged**